src/renderer/dx/DxRenderer.cpp

```diff
@@ -748,50 +774,26 @@ Microsoft::WRL::ComPtr<IDXGISwapChain1> DxEngine::GetSwapChain()
 // Return Value:
 // - S_OK
 [[nodiscard]] HRESULT DxEngine::InvalidateScroll(const COORD* const pcoordDelta) noexcept
+try
 {
-    if (pcoordDelta->X != 0 || pcoordDelta->Y != 0)
-    {
-        try
-        {
-            POINT delta = { 0 };
-            delta.x = pcoordDelta->X * _glyphCell.cx;
-            delta.y = pcoordDelta->Y * _glyphCell.cy;
+    RETURN_HR_IF(E_INVALIDARG, !pcoordDelta);
 
-            _InvalidOffset(delta);
+    const til::point deltaCells{ *pcoordDelta };
 
-            _invalidScroll.cx += delta.x;
-            _invalidScroll.cy += delta.y;
-
-            // Add the revealed portion of the screen from the scroll to the invalid area.
-            const RECT display = _GetDisplayRect();
-            RECT reveal = display;
-
-            // X delta first
-            OffsetRect(&reveal, delta.x, 0);
-            IntersectRect(&reveal, &reveal, &display);
-            SubtractRect(&reveal, &display, &reveal);
-
-            if (!IsRectEmpty(&reveal))
-            {
-                _InvalidOr(reveal);
-            }
+    if (deltaCells != til::point{ 0, 0 })
+    {
+        const til::point deltaPixels = deltaCells * _glyphCell;
 
-            // Y delta second (subtract rect won't work if you move both)
-            reveal = display;
-            OffsetRect(&reveal, 0, delta.y);
-            IntersectRect(&reveal, &reveal, &display);
-            SubtractRect(&reveal, &display, &reveal);
+        // Shift the contents of the map and fill in revealed area.
+        _invalidMap.translate(deltaCells, true);
 
-            if (!IsRectEmpty(&reveal))
-            {
-                _InvalidOr(reveal);
-            }
-        }
-        CATCH_RETURN();
+        // TODO: should we just maintain it all in cells?
+        _invalidScroll += deltaPixels;

```

Comment:
I like the idea of doing this in pixels, for the future idea of enabling scrolling a partial line into view, but right now this might be part of the cause of the horrible artifacts on high-DPI displays

Reply:
Hm OK.

Reply:
Had to do it in cells because pixels causes weird drift on high DPI.